ports/libsimpleservo/api/src/gl_glue.rs

```diff
@@ -77,10 +81,122 @@ pub mod egl {
+        let opengl_pointer;
+        opengl_pointer = unsafe {
+            gl::GlFns::load_with(|addr| {
+                let symbol_name: CFString = str::FromStr::from_str(addr).unwrap();
+                let framework_name: CFString = str::FromStr::from_str("com.apple.opengl").unwrap();
+                let framework =
+                    CFBundleGetBundleWithIdentifier(framework_name.as_concrete_TypeRef());
+                let symbol =
+                    CFBundleGetFunctionPointerForName(framework, symbol_name.as_concrete_TypeRef());
+                symbol as *const c_void
+            })
+        };
+        if opengl_pointer.is_null() {
+            Err("OpenGL isn't configured/installed")
+        }
+        info!("OpenGL loaded");
+        Ok(opengl_pointer)
+    }
+}
+
+#[cfg(target_os = "windows")]
+pub mod gl {
+    use servo::gl::GlType::Gl;
+    use std::ffi::CString;
+    use std::rc::Rc;
+    use winapi::um::libloaderapi::{GetProcAddress, LoadLibraryA};
+
+    pub fn init() -> Result<Rc<Gl>, &'static str> {
+        info!("Loading OpenGL...");
+
+        let opengl_pointer;
+        let dll = b"OpenGL32.dll\0" as &[u8];
+        let dll = unsafe { LoadLibraryA(dll.as_ptr() as *const _) };
+        if dll.is_null {
+            Err("Can't find Opengl32.dll, OpenGL isn't configured/installed")
+        } else {
+            unsafe {
+                opengl_pointer = gl::GlFns::load_with(|addr| {
+                    let addr = CString::new(addr.as_bytes()).unwrap();
+                    let addr = addr.as_ptr();
+                    GetProcAddress(dll, addr) as *const _
+                });
+            }
+        }
+        info!("OpenGL loaded");
+        Ok(opengl_pointer)
+    }
+}
+
+#[cfg(any(
+    target_os = "dragonfly",
+    target_os = "linux",
+    target_os = "freebsd",
+    target_os = "openbsd"
+))]
+extern crate glutin;
+
+pub mod gl {
+    use glutin::api::dlopen;
+    use glutin::api::glx::ffi::glx::Glx;
+    use servo::gl::GlType::Gl;
+    use std::ffi::CString;
+    use std::os::raw::c_void;
+    use std::rc::Rc;
+    pub fn init() -> Result<Rc<Gl>, &'static str> {
+        info!("Loading OpenGL");
+
+        pub mod ffi {
+            pub mod glx {
+                include!(concat!(env!("OUT_DIR"), "/glx_bindings.rs"));
+            }
+            pub mod glx_extra {
+                include!(concat!(env!("OUT_DIR"), "/glx_extra_bindings.rs"));
+            }
+        }
+
+        let mut opengl_dlopen =
+            unsafe { dlopen::dlopen(b"libGL.so.1\0".as_ptr() as *const _, dlopen::RTLD_NOW) };
+        if opengl_dlopen.is_null() {
+            unsafe {
+                opengl_dlopen = dlopen::dlopen(b"libGL.so\0".as_ptr() as *const _, dlopen::RTLD_NOW)
+            };
+        }
+
+        let opengl_pointer = ffi::glx::Glx::load_with(|addr| {
+            let addr_name = CString::new(addr).unwrap();
+            unsafe { ffi::glx::Glx::GetProcAddress(addr_name.as_ptr() as *const _) as *const _ }
```

Comment:
You can't just use `Glx` like that. You need to initialize it from the opened library.

See https://github.com/tomaka/glutin/blob/c16f997e4c6898bc3a3603c74c341c2a5497839c/src/platform/linux/x11.rs#L54

I haven't tested, but it should look like this:

```
let glx = Glx::load_with(|sym| {
  let sym = CString::new(sym).unwrap();
  unsafe { dlopen::dlsym(libglx, sym.as_ptr()) }
});

opengl_pointer = gl::GlFns::load_with(| addr|  glx.GetProcAddress(addr))
```